tests/analyses/test_checks.py

```diff
@@ -0,0 +1,97 @@
+from datetime import datetime
```

Comment:
I find the tests in this module unecessarily hard to read.

I think talking a class approach would be clearer. For each check function, you can have a class with two test methods: `test_ok`, `test_error`. The methods will end up being very obvious compared to the parametrized tests below.

Parametrization is used a lot in the API tests because they would be repetivitive if we didn't. The tests in this module are better without parametrization.